core/include/kernel/tcg.h

```diff
@@ -0,0 +1,164 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (c) 2022, Linaro Limited
+ *
+ * This file refers the following TCG specification.
+ * TCG PC Client Platform Firmware Profile Specification
+ */
+
+#ifndef __KERNEL_TCG_H__
+#define __KERNEL_TCG_H__
+
+#include <tee_api_types.h>
+#include <tpm2.h>
+
+#define TPM2_EVENT_LOG_SIZE		4096
+
+/*
+ * SHA1 Event Log Entry Format
+ *
+ * @pcr_index:  PCRIndex event extended to
+ * @event_type: Type of event (see EFI specs)
+ * @digest:     Value extended into PCR index
+ * @event_size: Size of event
+ * @event:      Event data
+ */
+struct tcg_pcr_event {
+	uint32_t pcr_index;
+	uint32_t event_type;
+	uint8_t digest[TPM2_SHA1_DIGEST_SIZE];
+	uint32_t event_size;
+	uint8_t event[];
+};
+
+/*
+ * Crypto Agile Log Entry Format
+ *
+ * @pcr_index:	PCRIndex event extended to
+ * @event_type:	Type of event
+ * @digests:	List of digestsextended to PCR index
+ * @event_size: Size of the event data
+ * @event:	Event data
+ */
+struct tcg_pcr_event2 {
```

Comment:
What does the `2` in the name signify?

Reply:
The name comes from the specification. Spec mentions SHA1 Event Log Entry Format as tcg_pcr_event. This was for devices with only single hash bank. This is still used to specify the header in event log.
So event log has
- event log header (of type tcg_pcr_event)
- spec ID event [ type tcg_efi_spec_id_event]
- multiple events of type tcg_pcr_event2 [ count of digest and hash algorithms in this needs to match what is parsed from spec ID event above)
-
For devices which support multiple hash banks, event log format entry was changed and was called tcg_pcr_event2.